saw-core-sbv/src/Verifier/SAW/Simulator/SBV.hs

```diff
@@ -722,6 +735,7 @@ newVars (FOTRec tm) = do
 newCodeGenVars :: (Natural -> Bool) -> FirstOrderType -> StateT Int IO (SBVCodeGen SValue)
 newCodeGenVars _checkSz FOTBit = nextId <&> \s -> (vBool <$> svCgInput KBool s)
 newCodeGenVars _checkSz FOTInt = nextId <&> \s -> (vInteger <$> svCgInput KUnbounded s)
+newCodeGenVars _checkSz (FOTIntMod _) = nextId <&> \s -> (vInteger <$> svCgInput KUnbounded s)
```

Comment:
Does is make sense to constrain the domain of this variable here?

Reply:
You mean, add a guard to enforce a nonzero modulus? We never use type `IntMod 0` in saw-core translated from cryptol, which doesn't allow `Z 0`; but I assumed that we'd treat the saw-core `IntMod 0` type as isomorphic to `Integer`. So the `n = 0` case would do exactly the same thing.

Reply:
No, I mean: should we constrain the fresh generated variable to be in the range `[0,n)`?  We do this in Cryptol and it seems to help the solvers in some instances.  For this exact case (code generation) I'm not so sure it matters, but for the other cases listed here it might.  And indeed, if we do that we'd need to make `n = 0` a special case.

Reply:
Yes, we probably should do that at some point. But I think we would need to add some new plumbing to support that; I don't think it's something we can add with just a local change, is it? If it requires other supporting changes, it's probably best left to a separate PR.

Reply:
Agreed, this is probably better done as a separate PR.